hack/aks/README.md

```diff
@@ -45,4 +45,66 @@ AKS Clusters
   windows-nodepool-up                       Add windows node pool
   down                                      Delete the cluster
   vmss-restart                              Restart the nodes of the cluster
+
+BYO CNI Automation
+  byocni-cluster-up                         Create complete BYO CNI cluster with CNS and CNI (default: Cilium)
+  deploy-cns                               Deploy CNS to the cluster  
+  deploy-cilium                            Deploy Cilium to the cluster
+  byocni-cluster-vars                      Show variables for BYO CNI cluster setup
+  validate-cni-type                        Validate the CNI type
+```
+
+## BYO CNI Cluster Automation
+
+The Makefile now includes automated setup for complete BYO CNI clusters with CNS and CNI deployment.
+
+### Quick Start
+
+Create a BYO CNI cluster with Cilium (default):
+```bash
+make byocni-cluster-up CLUSTER=my-cluster SUB=<subscription-id>
+```
+
+### Customization
+
+All parameters are configurable:
+```bash
+make byocni-cluster-up \
+    CLUSTER=my-cilium-cluster \
+    SUB=<subscription-id> \
+    CNS_VERSION=v1.6.0 \
+    CILIUM_DIR=1.16 \
+    CILIUM_VERSION_TAG=v1.16.5 \
+    CILIUM_IMAGE_REGISTRY=mcr.microsoft.com/containernetworking
+```
+
+### Available Configuration
+
+- `CNI_TYPE`: cilium (default) - Future CNI types can be added
```

Comment:
add option for azurecni. if azurecni specified, network-plugin should be azure and should remove network-dataplane cilium field